scalac-scoverage-runtime/js/src/main/scala/scalajssupport/NodeFile.scala

```diff
@@ -107,3 +110,11 @@ private[scalajssupport] object NodeFile extends JsFileObject {
     new NodeFile(path)
   }
 }
+
+private[scalajssupport] object NodeFile extends NodeLikeFile {
+  lazy val require = js.Dynamic.global.require
+}
+
+private[scalajssupport] object JSDOMFile extends NodeLikeFile {
+  lazy val require = js.Dynamic.global.Node.constructor("return require")()
+}
```

Comment:
This is how we break the JSDOM sandbox. See https://github.com/jsdom/jsdom/issues/2729.

Reply:
Nice catch and elegant solution! Thanks @armanbilge